File: UniqueItem.js
const Item = require("./Item");

class UniqueItem extends Item {
  constructor(name) {
    super(name);
    this.isLost = false;
  }

  setLost(lost) {
    this.isLost = lost;
  }

  getLost() {
    return this.isLost;
  }

  usedItem(message) {
    if (this.isLost) {
      console.log("Cet item est perdu, donc tu ne peux pas l'utiliser");
      return null;
    }
    console.log(`${this.getName()} a été utilisé. '${message}'`);
    return null;
  }
}

module.exports = UniqueItem;

File: UniqueItem.py
from Item import Item


class UniqueItem(Item):
    def __init__(self, name):
        super().__init__(name)
        self.is_lost = False

    def set_lost(self, lost):
        self.is_lost = lost

    def get_lost(self):
        return self.is_lost

    def used_item(self, message):
        if self.is_lost:
            print("Cet item est perdu, donc tu ne peux pas l'utiliser")
        else:
            print(f"{self.name} a été utilisé. '{message}'")
